styx-flyte-client/src/main/java/com/spotify/styx/flyte/client/FlyteInputsUtils.java

```diff
@@ -73,24 +76,30 @@ private FlyteInputsUtils() {
                 LOG.error(message);
                 throw new IllegalArgumentException(message);
               } else {
-                literalMapBuilder.putLiterals(key, buildLiteralForPartition(parameter));
+                var literal = buildLiteralForPartition(parameter);
+                literalMapBuilder.putLiterals(key, literal);
               }
             });
 
     return literalMapBuilder.build();
   }
 
-  private static Timestamp toTimestamp(String value) {
-    Instant instant;
-    try {
-      instant = ParameterUtil.parseDate(value);
-    } catch (DateTimeParseException ignored1) {
+  static Timestamp toTimestamp(String value) {
+    List<Function<String, Instant>> functions = List.of(
+        ParameterUtil::parseDate,
+        ParameterUtil::parseDateHour,
+        ParameterUtil::parseDateMonth,
+        ParameterUtil::parseDateYear,
+        Instant::parse
+    );
+    for (Function<String, Instant> function : functions) {
       try {
-        instant = ParameterUtil.parseDateHour(value);
-      } catch (DateTimeParseException ignored2) {
-        instant = Instant.parse(value);
-      }
+        Instant instant = function.apply(value);
```

Comment:
why apply all the functions here?

Reply:
She is going from less specific to more specific because we don't know which of this applies